From e69ce8381fdc89c5ec4de50346550b0e8527dfe8 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Wed, 5 Jun 2024 01:19:15 +0300
Subject: [PATCH 68/68] AI: Remove shared vision from pending war target once
 revolting

Make sure that there's no shared vision with the war target
once senate has been dissolved and war can really be declared.

See RM #701

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 ai/default/daidiplomacy.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ai/default/daidiplomacy.c b/ai/default/daidiplomacy.c
index 6ac46a7fb8..730923c34a 100644
--- a/ai/default/daidiplomacy.c
+++ b/ai/default/daidiplomacy.c
@@ -1443,6 +1443,10 @@ void dai_revolution_start(struct ai_type *ait, struct player *pplayer)
   struct ai_plr *data = def_ai_player_data(pplayer, ait);
 
   if (data->diplomacy.war_target != NULL) {
+    if (gives_shared_vision(pplayer, data->diplomacy.war_target)) {
+      remove_shared_vision(pplayer, data->diplomacy.war_target);
+    }
+
     dai_declare_war(ait, pplayer, data->diplomacy.war_target);
 
     data->diplomacy.war_target = NULL;
-- 
2.43.0

